geotrek/common/parsers.py

```diff
@@ -202,24 +206,33 @@ def set_value(self, dst, src, val):
 
     def parse_real_field(self, dst, src, val):
         """Returns True if modified"""
+        if hasattr(self.obj, dst):
+            if dst in self.m2m_fields or dst in self.m2m_constant_fields:
+                old = set(getattr(self.obj, dst).all())
+            else:
+                old = getattr(self.obj, dst)
         if hasattr(self, 'filter_{0}'.format(dst)):
             val = getattr(self, 'filter_{0}'.format(dst))(src, val)
         else:
             val = self.apply_filter(dst, src, val)
         if hasattr(self.obj, dst):
             if dst in self.m2m_fields or dst in self.m2m_constant_fields:
-                old = set(getattr(self.obj, dst).all())
                 val = set(val)
                 if dst in self.m2m_aggregate_fields:
                     val = val | old
-            else:
-                old = getattr(self.obj, dst)
```

Comment:
We need to retrieve `old` value BEFORE running `apply_filter` because this method could happen to change the value on the object (like in Aggregator)